contrib/epee/src/string_tools.cpp

```diff
@@ -112,81 +116,70 @@ namespace string_tools
   
   std::string num_to_string_fast(int64_t val)
   {
-    /*
-    char  buff[30] = {0};
-    i64toa_s(val, buff, sizeof(buff)-1, 10);
-    return buff;*/
     return boost::lexical_cast<std::string>(val);
   }
-	
-	
-    bool compare_no_case(const std::string& str1, const std::string& str2)
-	{
-		
-		return !boost::iequals(str1, str2);
-	}
-	//----------------------------------------------------------------------------
-	 std::string& get_current_module_name()
-	{
-		static std::string module_name;
-		return module_name;
-	}
-	//----------------------------------------------------------------------------
-	 std::string& get_current_module_folder()
-	{	
-		static std::string module_folder;
-		return module_folder;
-	}
-	
+  
+  bool compare_no_case(const std::string& str1, const std::string& str2)
+  {
+    return !boost::iequals(str1, str2);
+  }
+
+  std::string& get_current_module_name()
+  {
+    static std::string module_name;
+    return module_name;
+  }
+
+  std::string& get_current_module_folder()
+  {  
+    static std::string module_folder;
+    return module_folder;
+  }
+
 #ifdef _WIN32
   std::string get_current_module_path()
   {
-    char pname [5000] = {0};
-    GetModuleFileNameA( NULL, pname, sizeof(pname));
-    pname[sizeof(pname)-1] = 0; //be happy ;)
-    return pname;
+    std::wstring buf;
+    buf.resize(MAX_PATH);
+    do
+    {
+        unsigned int len = GetModuleFileNameW(NULL, &buf[0], static_cast< unsigned int >(buf.size()));
```

Comment:
This also returns `0` on error, I think it just loops until buff exceeds `65536` in that case. Perhaps an explicit check? We could `throw` I guess, although that changes the functionality a bit.